### PR TITLE
Resolve deprecation warnings

### PR DESCRIPTION
**Pull request by jsiirola**
## Summary/Motivation:
@jghouse88 noted [here](https://github.com/gmlc-dispatches/dispatches/issues/26#issuecomment-820639246) that Pyomo generates a slew of warnings when run with `pytest` (these have been there for quite a while, but nosetest suppresses them).  This PR removes a significant amount of deprecated functionality.

This also started resolving some issues with running `pytest` on Pyomo.

## Changes proposed in this PR:
- remove use of:
   - Logger.warn()
   - assertEquals()
   - assertNotEquals()
   - assertAlmostEquals()
   - assertNotAlmostEquals()
   - assertRaisesRegexp()
   - assertRegexpMatches()
- Convert deprecated regex escape strings to raw string literals
- Remove deprecated use of Enum.__contains__ with non-Enum types
- Import Iterable, Sized from collections.abc
- Rename Test* classes that are not TestCase subclasses

### Legal Acknowledgement

By contributing to this software project, I have read the [contribution guide](https://pyomo.readthedocs.io/en/stable/contribution_guide.html) and agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by jsiirola** [comment]
This PR depends on #1934 for the fix to `find_library()` (apparently the cache rebuild that highlighted the problem with `find_library()` also purged the older caches, so all other PRs now see this problem)

File: pyomo/opt/tests/base/test_solver.py [hunk 1]
```diff
@@ -24,18 +24,18 @@
 old_tempdir = TempfileManager.tempdir
 
 
-class TestSolver1(pyomo.opt.OptSolver):
+class DummySolver1(pyomo.opt.OptSolver):
```
Comment (mrmundt) [inline pyomo/opt/tests/base/test_solver.py:27]: Just a suggestion: I think it's more common to refer to these as "mock" objects in Python. I'm fine with `DummySolver` but we may want to consider `MockSolver`

File: pyomo/opt/tests/base/test_factory.py [hunk 2]
```diff
@@ -26,23 +26,23 @@
 old_tempdir = TempfileManager.tempdir
 
 
-class TestWriter(pyomo.opt.AbstractProblemWriter):
+class DummyWriter(pyomo.opt.AbstractProblemWriter):
```
Comment (mrmundt) [inline pyomo/opt/tests/base/test_factory.py:29]: Just a suggestion: I think it's more common to refer to these as "mock" objects in Python. I'm fine with `DummyWriter` but we may want to consider `MockWriter`

File: pyomo/solvers/tests/mip/test_factory.py [hunk 3]
```diff
@@ -36,23 +36,23 @@ def setUpModule():
 def tearDownModule():
     TempfileManager.tempdir = old_tempdir
 
-class TestWriter(AbstractProblemWriter):
+class DummyWriter(AbstractProblemWriter):
```
Comment (mrmundt) [inline pyomo/solvers/tests/mip/test_factory.py:39]: Just a suggestion: I think it's more common to refer to these as "mock" objects in Python. I'm fine with `DummyWriter` but we may want to consider `MockWriter`

File: pyomo/solvers/tests/mip/test_solver.py [hunk 4]
```diff
@@ -32,7 +32,7 @@ def tearDownModule():
     TempfileManager.tempdir = old_tempdir
 
 
-class TestSolver2(pyomo.opt.OptSolver):
+class DummySolver2(pyomo.opt.OptSolver):
```
Comment (mrmundt) [inline pyomo/solvers/tests/mip/test_solver.py:35]: Just a suggestion: I think it's more common to refer to these as "mock" objects in Python. I'm fine with `DummySolver` but we may want to consider `MockSolver`

**Review by mrmundt — APPROVED** [review]

**Pull request merged**